src/serializers/infer.rs

```diff
@@ -75,6 +76,13 @@ pub(crate) fn infer_to_python_known(
         }};
     }
 
+    let serialize_with_serializer = |value: &PyAny, is_model: bool| {
```

Comment:
I was sort of just trying to follow the pattern above with serialize_dict, not sure if this is the best way to do it, open to alternative suggestions.

Also, in principle, we could use something like this as a fallback for any not-otherwise-serializable object that has a __pydantic_serializer__ attribute of type SchemaSerializer. I didn't do that here as I didn't have any good use case in mind and it seemed unnecessary anyway. (And the current logic will always use the BaseModel path if it has a __pydantic_validator__ anyway, so maybe that's good enough.)